lib/compat/wordpress-6.9/client-assets.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Client Assets compatibility functions for WordPress 6.9.
+ *
+ * Hooks into `render_block_data` to detect blocks that support preloading
+ * during client side navigation (used by interactivity router). This is addressed in core in:
+ * https://github.com/WordPress/wordpress-develop/pull/10357
+ *
+ * @package gutenberg
+ */
+
+if ( version_compare( get_bloginfo( 'version' ), '6.9.0', '<' ) ) {
```

Comment:
Let's use the same approach here:

```suggestion
if ( ! method_exists( 'WP_Interactivity_API', 'add_client_navigation_support_to_script_module' ) ) {
```